proxy/http/HttpSM.cc

```diff
@@ -1616,14 +1617,11 @@ HttpSM::handle_api_return()
     }
 
     // We only follow 3xx when redirect_in_process == false. Otherwise the redirection has already been launched (in
-    // SM_ACTION_SERVE_FROM_CACHE or SM_ACTION_SERVER_READ).redirect_in_process is set before this logic if we need more direction.
+    // SM_ACTION_SERVER_READ).redirect_in_process is set before this logic if we need more direction.
     // This redirection is only used with the build_error_reponse. Then, the redirection_tries will be increased by
     // state_read_server_reponse_header and never get into this logic again.
-    if (enable_redirection && !t_state.redirect_info.redirect_in_process && is_redirect_required() &&
-        (redirection_tries <= t_state.txn_conf->number_of_redirections)) {
+    if (enable_redirection && !t_state.redirect_info.redirect_in_process && is_redirect_required()) {
       do_redirect();
-    } else if (redirection_tries > t_state.txn_conf->number_of_redirections) {
-      t_state.squid_codes.subcode = SQUID_SUBCODE_NUM_REDIRECTIONS_EXCEEDED;
```

Comment:
Moved this to `do_redirect()`, since that method makes a decision based upon this test.